geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java

```diff
@@ -179,6 +182,19 @@
    */
   private JGroupsReceiver jgroupsReceiver;
 
+  public static void setChannelReceiver(JChannel channel, Receiver r) {
+    try {
+      // Channel.setReceiver() won't set a new receiver unless the field is null but that
```

Comment:
Looking at `Channel.setReceiver()`, this comment is false. Dig:

```java
    public void setReceiver(Receiver r) {
        if(receiver != null && r != null)
            log.warn("%s: receiver already set", getAddress());
        receiver=r;
    }
```

Reply:
I'll fix that comment.

Reply:
ok